internal/profile/profile_test.go

```diff
@@ -0,0 +1,79 @@
+package profile_test
+
+import (
+	"testing"
+
+	gh_profile "github.com/pkg/profile"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cyberark/secretless-broker/internal/profile"
+)
+
+type stopper struct {
+	eventChan chan struct{}
+}
+
+func (s *stopper) Stop() {
+	s.eventChan <- struct{}{}
+}
+
+func TestProfile(t *testing.T) {
+	t.Run("New profile only allows cpu and memory types", func(t *testing.T) {
+		for _, profileType := range []string{"cpu", "memory"} {
+			profile, err := profile.New(profileType)
+
+			assert.NotNil(t, profile)
+			assert.Nil(t, err)
+		}
+
+		for _, profileType := range []string{"foo", "bar"} {
+			profile, err := profile.New(profileType)
+
+			assert.Nil(t, profile)
+
+			// Sanity check
+			assert.NotNil(t, err)
+			if err == nil {
+				t.Fail()
+			}
+
+			assert.Contains(t, err.Error(), "Invalid profile type")
+			assert.Contains(t, err.Error(), profileType)
+		}
+	})
+
+	t.Run("New profile delegates to profile imterface object", func(t *testing.T) {
+		startEventChan := make(chan struct{}, 10)
+		stopEventChan := make(chan struct{}, 10)
+
+		mockProfiler := func(profiles ...func(*gh_profile.Profile)) interface{ Stop() } {
+			startEventChan <- struct{}{}
+
+			return &stopper{
+				eventChan: stopEventChan,
+			}
+		}
+
+		profile, err := profile.NewWithOptions("cpu", mockProfiler)
+
+		// Sanity check
+		assert.Nil(t, err)
+		if err != nil {
+			t.Fail()
+		}
+
+		// Sanity check #2
+		assert.Equal(t, 0, len(startEventChan))
+		assert.Equal(t, 0, len(stopEventChan))
+
+		profile.Start()
+
+		assert.Equal(t, 1, len(startEventChan))
+		assert.Equal(t, 0, len(stopEventChan))
+
+		profile.Stop()
+
+		assert.Equal(t, 1, len(startEventChan))
+		assert.Equal(t, 1, len(stopEventChan))
+	})
+}
```

Comment:
Per our conversation earlier today:

> I definitely think “only use channels when you have to” is good advice

I think we don't need channels here.  All we're testing is that the `Start()` func we're delegating to does in fact get called (same with `Stop()`).   And this is definitely the right way to do this test. 

But instead of closing over `startEventChan` and `stopEventChan` to capture that info, you can close over simple boolean flags `startWasCalled` and `stopWasCalled` and then assert on them directly.